lib/gears/debug.lua

```diff
@@ -108,6 +112,10 @@ function debug.deprecate(see, args)
         end
     end
     debug.print_warning(msg .. ".\n" .. tb)
+
+    if awesome then
+        awesome.emit_signal("debug::deprecate", msg, see, args)
+    end
```

Comment:
In theory, `wibox` doesn't depend on `capi.awesome`, I just don't want to add dependencies that makes the code harder to use/test on it's own. For example, one day we might want to run some wibox in their own process to improve latency. For now that's mostly to make the tests shims simpler.